pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -196,12 +205,20 @@ func newReconciler(ofClient openflow.Client, ifaceStore interfacestore.Interface
 		ofClient:          ofClient,
 		ifaceStore:        ifaceStore,
 		lastRealizeds:     sync.Map{},
-		idAllocator:       newIDAllocator(),
+		idAllocator: newIDAllocator(),
 		priorityAssigners: priorityAssigners,
 	}
 	return reconciler
 }
 
+// RunIDAllocatorWorker runs worker that deletes rules from cache in idAllocator
+// asynchronously.
+func (r *reconciler) RunIDAllocatorWorker(stopCh <-chan struct{}) {
+	wait.Until(r.idAllocator.worker, time.Second, stopCh)
+
+	<- stopCh
```

Comment:
Yes missed the point that wait.until not started with go routine. Made sure the stopCh signal is caught only at one place.

Reply:
My point was the worker is never going to stop even stopCh is received.
`r.idAllocator.worker` will only exit when the queue is closed, otherwise it waits for item from queue forever.
Normally this "Run" function is written like this:
```
defer c.queue.ShutDown()
go wait.Until(r.idAllocator.worker, time.Second, stopCh)
<- stopCh
```
So that when stopCh catches stop signal, the queue is shutdown, then worker exits and no longer starts.

Reply:
Thanks for the explanation. It makes sense. I see the same usage in antrea controller, traceflow controller, node route controller etc. However, in `Run` of pkg/agent/controller/network_policy_controller for `controller.queue`, I do not see code for shutting down the queue. Wondering if I am missing something.

Reply:
Then it must be missing by mistake. Thanks for catching it. I will add it.

Reply:
I can take care of that with this PR if that's ok with you.

Reply:
Sounds good to me, thanks

Reply:
Done.